modules/mockk/src/jvmMain/kotlin/io/mockk/impl/recording/JvmSignatureValueGenerator.kt

```diff
@@ -62,4 +62,27 @@ class JvmSignatureValueGenerator(val rnd: Random) : SignatureValueGenerator {
                 } as T
             }
     }
+
+    /**
+     * Generates a Long value in Duration's safe millisecond range to avoid AssertionError.
+     * Uses wider millisecond range while avoiding the denormalized range.
+     * See: https://github.com/mockk/mockk/issues/1401
+     */
+    private fun generateSafeLong(): Long {
+        return if (rnd.nextBoolean()) {
```

Comment:
Why are we randomly generating a positive or a negative number?

Reply:
To cover Duration's valid ranges. 

Duration has two safe ranges separated by the denormalized zone, 
so we randomly pick between the negative range `[-MAX_MILLIS, -MAX_NANOS_IN_MILLIS]` and positive range `[MAX_NANOS_IN_MILLIS + 1, MAX_MILLIS]`.